### DISCUSSION TITLE
JSBSim Axes Reference

### DISCUSSION THREAD

**Discussion by aregis74** [JSBSim-Team/jsbsim#588] [General]
Hello everyone,

I trying to compute the forces and moments outputs on the body axis for wingboy, horizontal tail and vertica tail separately. The xml was built using a set of axes: DRAG, SIDE, LIFT, ROLL, PITCH, and YAW.  In the manual reference below shows that  DRAG, SIDE, LIFT means the aero forces resultant will be computed in the wind axes. So, how about the aero moments resultants? Will be also computed on the wind axes? Do I need to built transfer matrix at the end to have all forces and moments in the body axes?

regards,
André
 
![axes](https://user-images.githubusercontent.com/71350921/154299364-a4a20489-688c-4ab0-a5ed-62b9eac0769d.PNG)
https://jsbsim-team.github.io/jsbsim-reference-manual/mypages/user-manual-forces-and-moments

**Comment by seanmcleod** [comment]
Yes in general JSBSim supports `wind`, `stability` and `body` axes frames for both forces and moments in terms of their input via the model's xml file.

In terms of output, no matter how the inputs were specified, JSBSim makes all combinations available for output.

So you can retrieve the resultant force for each axis in each of the `wind`, `stability`, `body` axes frames and the same goes for the moments.

https://github.com/JSBSim-Team/jsbsim/blob/59c0bd909dbe7232e77ac9dc03eb65f959f16cf9/src/models/FGAerodynamics.cpp#L579-L596

**Reply by aregis74** [reply]
Thank you Sean, I got these outputs. But I am trying to make a cross-check computing the sumation of all cefficients (wingbody, horizontal tail and vertical tail) and comparing with the outputs you listed above. As I described before I am using a set of axis DRAG, SIDE, LIFT, ROLL, PITCH and YAW. All the forces(Fx, Fy and Fz) are consistents with outputs above (wind, body and stability). However, the moments sumation is consistent with the output of body axis. Shouldn't be consistent with wind axis?

**Comment by seanmcleod** [comment]
Are you taking into account that there are additional moments from the forces if the `current cg != AeroRP`?

**Reply by aregis74** [reply]
Yes. First I got the results considering current cg == AeroRP disregarding these additional moments and second current cg != AeroRP. In both situation, my moments  sumation are the same of the moments output of body axis, (moments/l-aero-lbsft, moments/m-aero-lbsft, moments/n-aero-lbsft).

**Comment by seanmcleod** [comment]
> So, how about the aero moments resultants? Will be also computed on the wind axes?

Ah, I think I see your confusion. So originally JSBSim allowed you to specify the forces in `wind` or `body`, but the moments specified by `PITCH`, `ROLL`, `YAW` were always assumed to be in the `body` frame.

Quite a while ago I then added support for all 3 axes frames, `stability`, `wind`, `body` to be used for forces or moments. The JSBSim documentation hasn't really been properly updated.

**Reply by aregis74** [reply]
Ah, great!! I am using python as interface and jsbsim version 1.1.8. Do you remember wich version is this one you supported for all 3 axes frame?

Thanks a lot

**Reply by seanmcleod** [reply]
Don't know the version off-hand, but it was quite a while ago, Feb 28 2018

https://github.com/JSBSim-Team/jsbsim/commit/eb996220167f38e3a3a46af11a4c31f7bc8360a3

**Reply by aregis74** [reply]
ok, thank again! I'll look for.

**Reply by bcoconni** [reply]
> Ah, great!! I am using python as interface and jsbsim version 1.1.8. Do you remember wich version is this one you supported for all 3 axes frame?

For the sake of completeness, stability axes have been added by @seanmcleod in early March 2018 (commits eb996220167f38e3a3a46af11a4c31f7bc8360a3 and 7a5b411671ae0f7a4723ae182a94fc5aa8245f96). So this feature is included in JSBSim since `v1.1.0`.

**Comment by seanmcleod** [comment]
Quite a while ago (Feb 28 2018 - https://github.com/JSBSim-Team/jsbsim/commit/eb996220167f38e3a3a46af11a4c31f7bc8360a3) I added support for all 3 axes frames, `stability`, `wind`, `body` to be used for forces or moments. The JSBSim documentation hasn't really been properly updated.

I think this is the current comprehensive list. Once double-checked, @bcoconni, @agodemar, @jonsberndt we should update the JSBSim documentation and maybe have a specific Wiki entry for it as well. 

### Forces

#### Body

```xml
<axis name="AXIAL|SIDE|NORMAL" />
<axis name="X|Y|Z" />
<axis name="X|Y|Z" frame="BODY" />
```

#### Wind

```xml
<axis name="DRAG|SIDE|LIFT" />
<axis name="X|Y|Z" frame="WIND" />
```

#### Stability
```xml
<axis name="X|Y|Z" frame="STABILITY" />
```

### Moments

#### Body

```xml
<axis name="ROLL|PITCH|YAW" />
<axis name="ROLL|PITCH|YAW" frame="BODY" />
```

#### Wind

```xml
<axis name="ROLL|PITCH|YAW" frame="WIND" />
```

#### Stability
```xml
<axis name="ROLL|PITCH|YAW" frame="STABILITY" />
```

**Comment by aregis74** [comment]
Great! I updated and all forces and moments matched now!
Thanks a lot!!!
